shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -466,6 +533,214 @@ fun NearbyStaticData.withRealtimeInfo(
         .sortedWith(compareBy(Route.relevanceComparator(pinnedRoutes)) { it.sortRoute() })
 }
 
+/**
+ * For reasons, for mid-line subway shuttles/suspensions, schedules will correctly reflect the
+ * closure and use route patterns that stop at temporary terminals, but predictions will use the
+ * canonical route patterns. The way we work around this is that for
+ * 1. subway routes
+ * 2. with alerts somewhere on the line
+ * 3. where the schedule has non-typical patterns instead of typical patterns
+ * 4. but the predictions all have typical patterns instead of non-typical patterns
```

Comment:
question: How does this perform is there is an alert on only a couple of stops on one of the RL branches?

Reply:
Correctly now, I think. (I'm beginning to appreciate the value of unit testing the individual helper functions.)